src/App.tsx

```diff
@@ -1,74 +1,153 @@
 import React from 'react';
 import './App.scss';
 import { peopleFromServer } from './data/people';
+import { Person } from './types/Person';
 
-export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
-
-  return (
-    <div className="container">
-      <main className="section is-flex is-flex-direction-column">
-        <h1 className="title" data-cy="title">
-          {`${name} (${born} - ${died})`}
-        </h1>
-
-        <div className="dropdown is-active">
-          <div className="dropdown-trigger">
-            <input
-              type="text"
-              placeholder="Enter a part of the name"
-              className="input"
-              data-cy="search-input"
-            />
-          </div>
+type State = {
+  query: string;
+  selectedPerson: Person | null;
+  isDropedDown: boolean;
+};
+
+export class App extends React.Component {
```

Comment:
OK, but I will not rewrite the component all the way to the class component, because it does not make sense to spend my time on that, as there is no error.